packages/contracts/contracts/Dependencies/TwapWeightedObserver.sol

```diff
@@ -95,8 +95,9 @@ contract TwapWeightedObserver is ITwapWeightedObserver {
             return virtualAvgValue;
         }
 
-        uint256 weightedAvg = data.lastObservedAverage * (PERIOD - futureWeight);
-        uint256 weightedVirtual = virtualAvgValue * (futureWeight);
+        uint256 weightedAvg = uint256(data.lastObservedAverage) *
+            (uint256(PERIOD) - uint256(futureWeight));
```

Comment:
`PERIOD` & `futureWeight` are defined as `uint256`, no need to explicitly cast for them